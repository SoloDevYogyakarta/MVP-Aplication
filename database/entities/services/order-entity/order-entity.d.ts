import { Model, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
export declare class OrderEntity extends BaseEntity {
    name: string;
    desc: string;
    user_id: number;
}
export type OrderCreationAttribute = Optional<OrderEntity, 'id'>;
export interface OrderInstance extends Model<OrderCreationAttribute, OrderEntity>, OrderEntity {
}
export declare const orderEntity: import("sequelize").ModelCtor<OrderInstance>;
