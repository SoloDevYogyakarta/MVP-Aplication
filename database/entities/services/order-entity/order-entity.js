"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.orderEntity = exports.OrderEntity = void 0;
const sequelize_1 = require("sequelize");
const base_entity_1 = require("../../base-entity/base-entity");
const entity_1 = require("../../entity");
class OrderEntity extends base_entity_1.BaseEntity {
}
exports.OrderEntity = OrderEntity;
exports.orderEntity = entity_1.sequelize.define('ORDER', {
    name: sequelize_1.DataTypes.STRING,
    desc: sequelize_1.DataTypes.TEXT,
    user_id: sequelize_1.DataTypes.INTEGER,
    createdAt: sequelize_1.DataTypes.DATE,
}, {
    tableName: 'ORDER',
    schema: 'PRODUCTS',
});
//# sourceMappingURL=order-entity.js.map