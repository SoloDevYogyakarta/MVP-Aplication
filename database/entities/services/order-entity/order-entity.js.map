{"version":3,"file":"order-entity.js","sourceRoot":"","sources":["../../../../../src/database/entities/services/order-entity/order-entity.ts"],"names":[],"mappings":";;;AAAA,yCAAwE;AACxE,+DAA2D;AAC3D,yCAAyC;AAEzC,MAAa,WAAY,SAAQ,wBAAU;CAG1C;AAHD,kCAGC;AAMY,QAAA,WAAW,GAAG,kBAAS,CAAC,MAAM,CACzC,OAAO,EACP;IACE,IAAI,EAAE,qBAAS,CAAC,IAAI;IACpB,OAAO,EAAE,qBAAS,CAAC,OAAO;CACO,EACnC;IACE,SAAS,EAAE,OAAO;IAClB,MAAM,EAAE,UAAU;CACnB,CACF,CAAC"}