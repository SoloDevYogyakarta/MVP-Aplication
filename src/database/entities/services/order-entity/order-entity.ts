import { nanoid } from 'nanoid';
import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class OrderEntity extends BaseEntity {
  user_id: string;
}

export type OrderCreationAttribute = Optional<OrderEntity, 'id'>;
export interface OrderInstance
  extends Model<OrderCreationAttribute, OrderEntity>,
    OrderEntity {}
export const orderEntity = sequelize.define<OrderInstance>(
  'ORDER',
  {
    public_id: DataTypes.STRING,
    user_id: DataTypes.STRING,
  } as ModelAttributes<OrderInstance>,
  {
    tableName: 'ORDER',
    schema: 'SERVICES',
    hooks: {
      beforeCreate(attributes, options) {
        const instance = attributes;
        instance.public_id = nanoid();
      },
    },
  },
);
