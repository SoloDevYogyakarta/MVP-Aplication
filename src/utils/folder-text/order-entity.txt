{"id":66,"user_id":"CUhFrjFaF5zHN4UFCq3Gj","updatedAt":"2023-11-05T12:41:37.328Z","createdAt":"2023-11-05T12:41:37.328Z","public_id":"D7fqsfHaY4DxFH9ZFtknb"}