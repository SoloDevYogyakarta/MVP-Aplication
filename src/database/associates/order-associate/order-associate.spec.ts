import { ModelCtor } from 'sequelize';
import { OrderInstance } from '../../../database/entities/services/order-entity/order-entity';
import {
  orderAssociate,
  orderAttribute,
  orderInclude,
} from './order-associate';

describe('OrderAssociate', () => {
  let id!: number;
  let entity: ModelCtor<OrderInstance>;

  beforeEach(() => {
    entity = orderAssociate;
  });

  it('should to be defined', () => expect(entity).toBeDefined());

  it('render correctly', () => expect(entity).toMatchSnapshot());

  if (id) {
    it('findOne with relationship', async () => {
      const findOne = await entity.findOne({
        where: { id },
        attributes: orderAttribute,
        include: orderInclude,
      });
      expect(findOne.id).toEqual(id);
    });

    it('findAll', async () => {
      const findAll = await entity.findAll({
        attributes: orderAttribute,
        include: orderInclude,
      });
      expect(findAll.length).not.toEqual(0);
    });
  }
});
