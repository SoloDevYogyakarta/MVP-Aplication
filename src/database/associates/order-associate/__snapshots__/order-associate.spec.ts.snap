// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderAssociate render correctly 1`] = `[Function]`;
